specification/inference/_types/Results.ts

```diff
@@ -57,13 +65,29 @@ export class TextEmbeddingResult {
   embedding: DenseVector
 }
 
+/**
+ * TextEmbeddingInferenceResult is an aggregation of mutually exclusive text_embedding variants
+ * @variants container
+ */
+export class TextEmbeddingInferenceResult {
+  text_embedding_bytes?: Array<TextEmbeddingByteResult>
+  text_embedding?: Array<TextEmbeddingResult>
+}
+
 /**
  * The completion result object
  */
 export class CompletionResult {
   result: string
 }
 
+/**
+ * Defines the completion result.
+ */
+export class CompletionInferenceResult {
```

Comment:
I'm open to other ideas for naming the classes. `*Result` was already taken for everything for the nested field which is why I went with `*InferenceResult`.